modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/SPATopHeadJSPDynamicInclude.java

```diff
@@ -101,19 +101,15 @@ public void include(
 			"validStatusCodes", _spaHelper.getValidStatusCodesJSONArray()
 		);
 
-		String initModuleName = _npmResolver.resolveModuleName(
-			"frontend-js-spa-web/init");
+		StringBundler javascriptSB = new StringBundler(3);
 
-		ScriptData initScriptData = new ScriptData();
+		javascriptSB.append("frontendJsSpaWebInit.default(");
+		javascriptSB.append(configJSONObject.toJSONString());
+		javascriptSB.append(");");
 
-		initScriptData.append(
-			null,
-			"frontendJsSpaWebInit.default(" + configJSONObject.toJSONString() +
-				")",
-			initModuleName + " as frontendJsSpaWebInit",
-			ScriptData.ModulesType.ES6);
-
-		initScriptData.writeTo(httpServletResponse.getWriter());
+		_jsModuleLauncher.writeScript(
+			httpServletResponse.getWriter(), "frontend-js-spa-web",
+			"frontendJsSpaWebInit", javascriptSB.toString());
```

Comment:
Similar to https://github.com/liferay-frontend/liferay-portal/pull/696/files#r561020549:
- The `frontend-js-spa-web` is the `name` of the module as per its `package.json`
- How's `frontendJSSpaWebInit` being generated? In this case, it seems reasonable that it is `normalize(frontend-js-spa-web) + stripExtension(packageJson.main)`. Is that what webpack federation is generating?

Reply:
It's not being generated, it's the name/alias we want to give it. I used the previous existing names, but I can rename them if we want, as it seems it leads to confusion.

Reply:
I preferred not to use automatic name generation like in `ScriptData` because:

1) it's only one var
2) that strategy leads to several hours being wasted writing documentation, looking for it, and trying to remember how the magic variables are generated (and failing, of course :joy: )

Reply:
If the name is meaningless, we shouldn't ask for one, since it seems completely irrelevant.

Reply:
Just do `StringUtil.randomId()` and be done with it :D

Reply:
No, it's not meaningless, it's something you will use in your script code (in the next parameter). So, since it's your code, it makes sense that you decide how to name it.

Reply:
Another option would be using a fixed name but then you restrict the user code because you make that name unusable (kind of a reserved word). And still you need to document the API, otherwise, first time users won't know what to put in the JS block to refer to the required module.